### PR TITLE
feat: adding the report feature for snaps

### LINKED ISSUE #1490: Report link on App Page
Screenshot 2023-11-14 at 14 25 44

### PR DESCRIPTION
**Pull request by BLKKKBVSIK**
In this PR:

- Adding the UI and the necessary functionality to be able to report a Snap. It use the same principle as snapcraft.io

There was no Figma file, so I improvised some visual aspects to be as close as possible from snapcraft.io, but with Yaru theming.

Screenshot 2023-12-26 at 13 50 39

fix #1490

### CODE REVIEW

File: packages/app_center/lib/src/snapd/snap_report.dart [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = AppLocalizations.of(context);
+    final textTheme = Theme.of(context).textTheme;
+    final titleTextStyle = textTheme.headlineSmall!;
+
+    return Dialog(
+      child: Container(
+        padding: const EdgeInsets.all(16.0),
```
Comment (spydon) [inline packages/app_center/lib/src/snapd/snap_report.dart:31]: Is there no theming for padding sizes?

File: packages/app_center/pubspec.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 name: app_center
 description: App Center
 version: 1.0.0
-publish_to: 'none'
+publish_to: "none"
 
 environment:
-  sdk: '>=3.1.0 <4.0.0'
+  sdk: ">=3.1.0 <4.0.0"
```
Comment (spydon) [inline packages/app_center/pubspec.yaml:7]: Isn't the default ' in `flutter create`?

**Review by spydon — COMMENTED** [review]
A lot of hard coded sizes, but maybe it's like that all over this app?

File: packages/app_center/pubspec.yaml (hunk 2 above)
Comment (BLKKKBVSIK) [reply packages/app_center/pubspec.yaml:7]: Got no idea when it got modified to be honest, definitively not intended.

**Comment by anasereijo** [comment]
awesome! just some tiny things (I hope)

- can we have the cancel button as an outlined button? 
- the form headings are a bit too close to the fields, is this something we can improve or is it what we have on yaru?
- what's the width of the popup? Seems to be a bit wide, perhaps we can make it smaller. But if it's a yaru default never mind.

**Comment by BLKKKBVSIK** [comment]
Screenshot 2024-01-08 at 09 12 46

**Review by spydon — APPROVED** [review]

**Comment by BLKKKBVSIK** [comment]
@anasereijo If that's OK for you, I'll merge the PR.

**Review by anasereijo — APPROVED** [review]

**Comment by d-loose** [comment]
Since I was just cleaning up some code coverage related things, I noticed that there aren't any tests for this feature - It'd be great if we could add some! :)

**Pull request merged**